From: Justin Carpentier <justin.carpentier@inria.fr>
Date: Sun, 10 Nov 2019 08:23:17 +0100
Subject: defs: use noexcept only for C++11 and more

---
 include/assimp/defs.h | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/include/assimp/defs.h b/include/assimp/defs.h
index 05a5e3f..205504a 100644
--- a/include/assimp/defs.h
+++ b/include/assimp/defs.h
@@ -298,7 +298,11 @@ static const ai_real ai_epsilon = (ai_real) 0.00001;
 #define AI_MAX_ALLOC(type) ((256U * 1024 * 1024) / sizeof(type))
 
 #ifndef _MSC_VER
-#  define AI_NO_EXCEPT noexcept
+#  if __cplusplus >= 201103L // C++11
+#    define AI_NO_EXCEPT noexcept
+#  else
+#    define AI_NO_EXCEPT
+#  endif
 #else
 #  if (_MSC_VER >= 1915 )
 #    define AI_NO_EXCEPT noexcept
